src/other/tables.jl

```diff
@@ -0,0 +1,39 @@
+using Tables, IteratorInterfaceExtensions
+
+Tables.istable(::Type{<:AbstractDataFrame}) = true
+Tables.columnaccess(::Type{<:AbstractDataFrame}) = true
+Tables.columns(df::AbstractDataFrame) = df
+Tables.rowaccess(::Type{DataFrame}) = true
+Tables.rows(df::DataFrame) = eachrow(df)
+
+Tables.schema(df::AbstractDataFrame) = Tables.Schema(names(df), eltypes(df))
+Tables.schema(df::DFRowIterator) = Tables.schema(df.df)
+
+getvector(x::AbstractVector) = x
+getvector(x) = collect(x)
+fromcolumns(x) = DataFrame(Any[getvector(c) for c in Tables.eachcolumn(x)], collect(propertynames(x)))
+
+function DataFrame(x::T) where {T}
+    if Tables.istable(T)
+        return fromcolumns(Tables.columns(x))
+    end
+    y = IteratorInterfaceExtensions.getiterator(x)
+    yT = typeof(y)
+    if Base.isiterable(yT)
+        Base.depwarn("constructing a DataFrame from an iterator is deprecated; $T should support the Tables.jl interface", nothing)
```

Comment:
I thought the goal was to converge towards a single common interface in Tables.jl?

Anyway it's going to be hard or impossible to support two interfaces given that we cannot dispatch on a particular type: this method will be called as a fallback for all kinds of tables.

Reply:
I agree that we don't need to be so hasty in deprecating, especially with Tables.jl not even used anywhere yet. I'll comment out the depwarn and we can see how things play out over the next little while; it would certainly be annoying for lots of users to see that depwarn without anyway to get rid of it.